streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -162,17 +162,19 @@
      *         |             v
      *         |       +-----+--------+
      *         +-----> | Pending      |
-     *                 | Shutdown     |
-     *                 +-----+--------+
-     *                       |
-     *                       v
-     *                 +-----+--------+
-     *                 | Not Running  |
+     *         |       | Shutdown     |
+     *         |       +-----+--------+
+     *         |             |
+     *         |             v
+     *         |       +-----+--------+
+     *         +-----> | Not Running  |
      *                 +--------------+
      * </pre>
+     * Note the following:
+     * - Any state can go to PENDING_SHUTDOWN (during clean shutdown) or NOT_RUNNING (e.g., during an exception).
      */
     public enum State {
-        CREATED(1, 2, 3), RUNNING(2, 3), REBALANCING(1, 2, 3), PENDING_SHUTDOWN(4), NOT_RUNNING;
+        CREATED(1, 2, 3), REBALANCING(1, 2, 3, 4), RUNNING(1, 3, 4), PENDING_SHUTDOWN(4), NOT_RUNNING;
```

Comment:
I have a comment about how any state can go to NOT_RUNNING anytime there is an exception that happens in that state. I'll add a note on rebalancing too.